pubmed_parser/pubmed_oa_parser.py

```diff
@@ -87,6 +87,18 @@ def parse_article_meta(tree):
     return dict_article_meta
 
 
+def parse_date(tree, date_type):
+    """Parse ppub and epub dates."""
+    pub_year_node = tree.find(f".//pub-date[@pub-type='{date_type}']/year")
+    pub_year = pub_year_node.text if pub_year_node is not None else ""
+    pub_month_node = tree.find(f".//pub-date[@pub-type='{date_type}']/month")
+    pub_month = pub_month_node.text if pub_month_node is not None else "01"
+    pub_day_node = tree.find(f".//pub-date[@pub-type='{date_type}']/day")
+    pub_day = pub_day_node.text if pub_day_node is not None else "01"
+
+    return {'year': pub_year, 'month': pub_month, 'day': pub_day}
```

Comment:
I think we should do the formatting here, not later. Any try to avoid a trailing "-".

Reply:
The problem is that [here](https://github.com/nils-herrmann/pubmed_parser/blob/13522a3e52666025f249ca1335aa993785ad8a96/pubmed_parser/pubmed_oa_parser.py#L239) we need the year. If we create the string in the function I would have to parse it afterwards.

Reply:
Not necessarily. You can also use just the first 4 characters of the full date string. But please turn it to integer.

Reply:
I changed the code to:
- avoid trailing "-" in the date string
- format the date before setting the values in the dictionary
- return the publication year as int (or None)